BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -404,6 +411,19 @@ protected virtual HttpRequestMessage CreateRequestMessage(Uri address, Stream co
                 request.Content.Headers.Add(ContentEncodingHeader, this.ContentEncoding);
             }
 
+            if (this.CredentialEnvelope != null)
+            {
+                // TODO: NEED TO USE CACHING
+                var aadToken = this.CredentialEnvelope.GetToken();
+
+                if (aadToken == null)
+                {
+                    // TODO: DO NOT SEND. RETURN FAILURE AND LET CHANNEL DECIDE WHEN TO RETRY.
```

Comment:
this will be considered a client side failure, and data would not be retried by default.
Its probably a complex part to decide if we want to store items for later retry, if the AAD is down..